src/core/mempool.rs

```diff
@@ -504,11 +538,72 @@ impl MemPoolDB {
         ))
     }
 
+    fn get_next_timestamp_at_next_block<E>(
+        &self,
+        chainstate: &mut StacksChainState,
+        mut curr_tip_consensus_hash: ConsensusHash,
+        mut curr_tip_block_hash: BlockHeaderHash,
+        mut curr_tip_height: Option<u64>,
```

Comment:
Interesting and curious to hear about other opinions, I'm just sharing a practice that I personally prefer (coming from habits from other languages).
I usually try to avoid mutable arguments. In this case, the arguments are marked as mutable so that the inner loop can update them for next iterations, I think I would make some local, mutable copies of the arguments instead - in this case, it looks fine, since this inputs are owned, and not mutable references: thank you Rust :).

Reply:
I agree, I also try to avoid mutable arguments and prefer passing any mutated values back explicitly (which is what I did here) - only did it like this because these inputs are not references.